endoreg_db/services/video_import.py

```
@@ -563,48 +904,75 @@ def _perform_frame_cleaning(self, FrameCleaner, processor_roi, endoscope_roi):
-                    cleaned_video_path.name,
-                    ContentFile(f.read())
-                )
-            self.current_video.save()
+        # Optional: enrich metadata using TrOCR+LLM on one random extracted frame
+        try:
+            # Prefer frames belonging to this video (UUID in path), else pick any frame
+            frame_candidates = list(RAW_FRAME_DIR.rglob("*.jpg")) + list(RAW_FRAME_DIR.rglob("*.png"))
+            video_uuid = str(self.current_video.uuid)
+            filtered = [p for p in frame_candidates if video_uuid in str(p)] or frame_candidates
+            if filtered:
+                sample_frame = random.choice(filtered)
```

Ruff (0.14.0)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)